server/backend/api/app/routers/router_guards.py

```diff
@@ -50,38 +41,63 @@ async def get_current_requester(
     request_cognito_user_id: str = Depends(get_request_cognito_user_id),
     access_roles: List[str] = Depends(get_access_roles),
     db: Session = Depends(database.get_db),
-):
-    LOGGER.debug(
-        f"Debug router_guard get_current_requester, current request_cognito_user_id: {request_cognito_user_id}"
-    )
+) -> Requester:
     LOGGER.debug(
-        f"Debug router_guard get_current_requester, current db connection: {db}"
+        f"Retrieving current requester from: request_cognito_user_id: {request_cognito_user_id}"
     )
-    fam_user: FamUser = crud_user.get_user_by_cognito_user_id(
+    fam_user: FamUser = crud_user.fetch_initial_requester_info(
         db, request_cognito_user_id
     )
     LOGGER.debug(
-        f"Debug router_guard get_current_requester, current fam_user: {fam_user}"
+        f"Crrent retrieved fam_user: {fam_user}"
     )
+
     if fam_user is None:
         utils.raise_http_exception(
             error_msg="Requester does not exist, action is not allowed.",
             error_code=ERROR_CODE_REQUESTER_NOT_EXISTS,
             status_code=HTTPStatus.FORBIDDEN,
         )
 
-    requester = Requester.model_validate(fam_user)
-    LOGGER.debug(
-        f"Debug router_guard get_current_requester, current requester: {requester}"
+    else:
+        custom_fields = _parse_custom_requester_fields(fam_user)
+        requester = Requester.model_validate(
+            {
+                **fam_user.__dict__,  # base db 'user' info
+                'access_roles': access_roles,  # role from JWT
+                **custom_fields  # build/conver custom attributes
+            }
+        )
+        LOGGER.debug(f"Current request user (requester): {requester}")
+        return requester
+
+
+def _parse_custom_requester_fields(fam_user: FamUser):
+    """
+    Conversation helper function to parse information from FamUser for some
+    custom attributes needed at Requester.
+    :fam_user: fetched FamUser from db with joined table information.
+    :return: dictionary contains custom attributes information for setting 'Requester'
+    """
+    user_type_code = fam_user.user_type_code
+    is_delegated_admin = len(fam_user.fam_access_control_privileges) > 0
+    has_current_terms_conditions_accepted = (
+        fam_user.fam_user_terms_conditions and
+        fam_user.fam_user_terms_conditions.version == CURRENT_TERMS_AND_CONDITIONS_VERSION
+    )
+    requires_accept_tc = (
+        user_type_code == UserType.BCEID and is_delegated_admin
+        and not has_current_terms_conditions_accepted
     )
-    requester.access_roles = access_roles
-    LOGGER.debug(f"Current request user (requester): {requester}")
-    return requester
+
+    return {
+        "is_delegated_admin": is_delegated_admin,
+        "requires_accept_tc": requires_accept_tc
+    }
```

Comment:
New custom attributes for "Requester" schema. The information is parsed from previous joined tables information from specific user.